src/dag/dag.cpp

```diff
@@ -443,9 +442,9 @@ void DagManager::addToDag(blk_hash_t const &hash, blk_hash_t const &pivot, std::
 
 bool DagManager::getLatestPivotAndTips(blk_hash_t &pivot, std::vector<blk_hash_t> &tips) const {
```

Comment:
These ref arguments dont make much sense because getFrontier() is returning copies anyway. I think it should return something like:
`std::optional<std::pair<blk_hash_t, std::vector<blk_hash_t>>>`
?

Reply:
good point I will change this